sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -409,8 +410,25 @@ object JdbcUtils extends Logging {
         row.setFloat(pos, rs.getFloat(pos + 1))
 
     case IntegerType =>
-      (rs: ResultSet, row: InternalRow, pos: Int) =>
-        row.setInt(pos, rs.getInt(pos + 1))
+      (rs: ResultSet, row: InternalRow, pos: Int) => {
+        val columnIdx = pos + 1
+        // SPARK-33888 - sql TIME type represents as physical int in millis
+        // Represents a time of day, with no reference to a particular calendar,
+        // time zone or date, with a precision of one millisecond.
+        // It stores the number of milliseconds after midnight, 00:00:00.000.
+        if (rs.getMetaData.getColumnType(columnIdx) == java.sql.Types.TIME) {
```

Comment:
do we run the `if` check for every input row?

Reply:
Yes. You're right.

Originally, I  made changes to private method `makeGetters` and pass in the `ResultSetMetadata` as additional parameter. Then we have a separate pattern matching for `case IntegerType if rsMetadata.getColumnType(columnIdx) == TIME =>`  But right now, the column index/position is not accessible at the pattern matching level, only available at the closure function :(

Any advice is deeply appreciated here.

Reply:
Actually, let me see if I can work with the `metadata` from `getSchema` function.